### DISCUSSION TITLE
Do the Master and Chunk servers have to be the same architecture?

### DISCUSSION THREAD

**Discussion by Qiangong2** [moosefs/moosefs#550] [Q&A]
## Have you read through available documentation, open Github issues and Github Ideas Discussions?

Yes

## Describe your request:

This is more just a general question. Would we be able to run a master server with an ARM64 chip and the chunkservers as x86? Or do all servers need to be the same architecture?

**Comment by eleaner** [comment]
I run master on x86 and some of chunkservers on ARM6

**Comment by XingangL** [comment]
I don't think it's necessary. MooseFS is a set of network storage applications that support running on heterogeneous platforms.

**Comment by chogata** [comment]
It's absolutely not necessary to have the same architecture, you can have the components on various machines and various OSes too. The only thing we recommend is for the chunk servers not to have drastic differences, when it comes to amount of space on hard drives and general computing power, but this is only because of performance, not any problems with data, and generally concerns MooseFS instances with a lot of active i/o. If your instance is more of a backup one, then really, everything goes when it comes to hardware ;)
